src/api/libcellml/enums.h

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright libCellML Contributors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+namespace libcellml {
+
+/**
+ * @brief The CellML element enum class.
+ *
+ * This enum describes CellML elements.
+ * A CellML element identifies elements in
+ * the underlying CellML specification format.
+ * The enum also adds an UNDEFINED entry for
+ * not specifying an underlying CellML element.
+ */
+enum class CellMLElement
+{
+    COMPONENT,
+    COMPONENT_REF,
+    CONNECTION,
+    ENCAPSULATION,
+    IMPORT,
+    MAP_VARIABLES,
+    MATHML,
+    MODEL,
+    RESET,
+    RESET_VALUE,
+    TEST_VALUE,
+    UNDEFINED,
+    UNIT,
+    UNITS,
+    VARIABLE,
+};
```

Comment:
Will we add `XML` and `INVALID` to this list so it can be used elsewhere?

Reply:
Well, no I hope not.  The XML one is a problem but it is not a CellML element.  The `INVALID` should be covered by `UNDEFINED` no?

Reply:
Ah, I thought yesterday you wanted an invalid type.  I'm fine with undefined to cover both.

Reply:
We still need to rename this so that it includes `XML` and is named `ItemType` so that it can fit with the rest of the code... That will have knock-on effects throughout the rest of this PR.